gmp/gvm_connection.py

```diff
@@ -959,23 +961,34 @@ def __init__(self, **kwargs):
                                    '/usr/local/var/run/gvmd.sock')
         self.shell_mode = kwargs.get('shell_mode', False)
         self.timeout = kwargs.get('timeout', 60)
+        self.read_timeout = kwargs.get('read_timeout', 30)
         self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM) #pylint: disable=E1101
         self.sock.settimeout(self.timeout)
         self.sock.connect(self.sockpath)
 
     def readAll(self):
+        self.first_element = None
+        self.parser = etree.XMLPullParser(('start', 'end'))
         response = ''
-        while True:
-            data = self.sock.recv(BUF_SIZE)
-            # Todo: Why does the sleep helps here? Sometimes it will break
-            # here because the message is missing some bytes at the end.
-            # Same script and with tls or ssh, then it works flawless without
-            # "sleep()"
-            time.sleep(0.000001)
-            response += data.decode()
+
+        read_timeout = self.read_timeout
+        break_timeout = time.time() + read_timeout
+        old_timeout = self.sock.gettimeout()
+        self.sock.settimeout(5)
+
+        while time.time() < break_timeout:
+            data = b''
+            try:
+                data = self.sock.recv(BUF_SIZE)
+            except (socket.timeout) as exception:
+                logger.debug('Warning: No data recieve from server: {0}'.format(exception))
+            self.parser.feed(data)
```

Comment:
Is this valid if a timeout has occurred? Wouldn't it be better to add a continue statement into the except case?